seed/php-sdk/unions/src/Types/Types/UnionWithLiteral.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Seed\Types\Types;
+
+use Seed\Core\Json\JsonProperty;
+use Seed\Core\Json\JsonSkip;
+use Seed\Core\Types\DiscriminatedUnion;
+use Seed\Core\Types\Discriminant;
+
+class UnionWithLiteral extends DiscriminatedUnion
+{
+    /**
+     * @var 'base' $base
+     */
+    #[JsonProperty('base')]
+    public string $base;
+
+    /**
+     * @var 'fern'|'_unknown' $type 
+     */
+    #[JsonProperty('type')]
+    #[Discriminant(['fern' => 'fern'])]
+    public string $type;
+
+    /**
+     * @var 'fern'|mixed
+     */
+    #[JsonSkip]
+    public mixed $value;
+
+    /**
+     * @param ?array{
+     *   base?: 'base',
+     *   type?: 'fern'|'_unknown',
+     *   value?: 'fern'|mixed,
+     * } $options
+     */
+    public function __construct(
+        private readonly ?array $options = null,
+    ) {
+        if (is_null($this->options)) {
+            throw new \Exception("Missing all required properties");
+        }
+        if (array_key_exists('base', $this->options)) {
+            $this->base = $this->options['base'];
+        } else {
+            throw new \Exception("Missing required property 'base'");
+        }
+        $this->type = $this->options['type'] ?? '_unknown';
+        $this->value = $this->options['value'] ?? null;
+    }
+
+    public static function fern(): UnionWithLiteral
+    {
+        return new UnionWithLiteral([
+            'base' => 'base',
+            'type' => 'fern',
+            'fern' => 'fern'
+        ]);
+    }
+
+    public static function _unknown(
+        mixed $_unknown
+    ): UnionWithLiteral {
+        return new UnionWithLiteral([
+            '_unknown' => $_unknown
+        ]);
+    }
+
+    public function asFern(): string
+    {
+        if ($this->type != 'fern') {
+            throw new \Exception(
+                "Expected type to be 'fern'; got '$this->type.'"
+            );
+        }
+
+        if ($this->value !== 'fern') {
+            throw new \Exception(
+                "Expected value to be \"fern\"."
+            );
+        }
```

Comment:
Depending on the phpstan literal support and whether or not we make this _always_ be the literal value, we might not need this check.

With that said, we should definitely validate against literal values in the [de]serialization layer. For example, if this is a response type and the server sends the following, we should fail to deserialize the type:

```php
{
  "fern": "something"
}
```

All that to say - the more we can move this validation logic to the serialization layer the better. That way when users have a constructed instance of the type, it's guaranteed to be valid.